src/doc/performance-tuning.md

```diff
@@ -110,95 +205,184 @@ Node 1 HugePages_Total:    32
+size is 512 packets and the maximum is 65536.  The NIC will fill this
+buffer with packets as it receives them: first to slot 0, then to slot
+1, all the way up to slot 511 (for a ring buffer of the default size),
+then back to slot 0, then slot 1, and so on.  Snabb will periodically
+take some packets out of this read buffer (currently 128 at a time),
+process them, then come back and take some more: wash, rinse, repeat.
+
+Many network functions offer the ability to configure the ring buffer
+size via a `--ring-buffer-size` argument.  (Again, if the network
+function you are using does not provide this option, file a bug.)
+What is the right size?  Well, there are a few trade-offs.  If the
+buffer is too big, it will take up a lot of memory and start to have
+too much of a cache footprint.  The ring buffer is mapped into Snabb's
+memory as well, and the NIC arranges for the ring buffer elements that
+it writes to be directly placed in L3 cache.  This means that
+receiving packets can evict other entries in L3 cache.  If your ring
+buffer is too big, it can evict other data resident in cache that you
+might want.
+
+Another down-side of having a big buffer is latency.  The bigger your
+buffer, the more the bloat.  Usually, Snabb applications always run
+faster than the incoming packets, so the buffer size isn't an issue when
+everything is going well; but in a traffic spike where packets come in
+faster than Snabb can process them, the buffer can remain full the whole
+time, which just adds latency for the packets that Snabb does manage to
+process.
+
+However, too small a buffer exposes the user to a higher risk of
+packet loss due to jitter in the Snabb breath time.  A "breath" is one
+cycle of processing a batch of packets, and as we mentioned it is
+currently up to 128 packets at a time.  This processing doesn't always
+take the same amount of time: the contents of the packets can
+obviously be different, causing different control flow and different
+data flow.  Even well-tuned applications can exhibit jitter in their
+breath times due to differences between what data is in cache (and
+which cache) and what data has to be fetched from memory.  Infrequent
+events like reconfiguration can also cause one breath to take longer
+than another.  Poorly-tuned applications might have other sources of
+latency such as garbage collection, though this is not usually the
+case in production-ready Snabb network functions like the NFV virtual
+switch or the lwAFTR.
+
+So, a bigger ring buffer insulates packet processing from breath
+jitter.  You want your ring buffer to be big enough to not drop
+packets due to jitter during normal operation, but not bigger than
+that.  In our testing we usually use the default ring buffer size.  In
+your operations you might want to increase this up to 2048 entries.
+We have not found that bigger ring buffer sizes are beneficial, but it
+depends very much on the environment.
+
```

Comment:
I've heard contradictory things on the benefits of 512 vs 1024 - the previous version strongly stumped for 1024. It might be worth explicitly preserving the information that some people have found 1024 to be a win compared to 512.